src/chainstate/stacks/index/storage.rs

```diff
@@ -891,6 +892,16 @@ impl TrieFileStorage {
     pub fn open_block(&mut self, bhh: &BlockHeaderHash, readwrite: bool) -> Result<(), Error> {
         let block_fork_ptr = self.fork_table.get_fork_ptr(bhh)?;
 
+        let sentinel = TrieFileStorage::block_sentinel();
+        if *bhh == sentinel {
```

Comment:
After my first pass through the diff, I'm still confused as to why this is needed?  To use a MARF, you first need to instantiate its on-chain state with `begin()`, and persist the changes with `commit()`.  The `begin()` method will call ultimately `open_block()` with a _non-sentinel_ block hash.

Reply:
the problem is when `get()` is called on a MARF that isn't in a transaction. the "open block" in the file storage is the sentinel, so when `get` tries to restore the previously opened block at the end of the function, it would lead to a corrupted TrieFileStorage

Reply:
Ah, okay.  Sounds good then :+1: